src/common/feedGenerator.ts

```diff
@@ -670,14 +700,44 @@ export const bookmarksFeedBuilder = (
       `bookmark."postId" = ${alias}.id AND bookmark."userId" = :userId`,
       { userId: ctx.userId },
     );
+
   if (unreadOnly) {
     newBuilder = newBuilder.andWhere((subBuilder) =>
       whereUnread(ctx.userId, subBuilder, alias),
     );
   }
-  if (listId && ctx.premium) {
-    newBuilder = newBuilder.andWhere('bookmark.listId = :listId', { listId });
+
+  const [firstFolderSubQuery, params] = getFirstFolderSubQuery({
+    builder: newBuilder,
+    userId: ctx.userId!,
+  });
+
+  if (ctx.isPlus) {
+    if (listId) {
+      newBuilder = newBuilder.andWhere('bookmark.listId = :listId', { listId });
+    } else {
+      newBuilder = newBuilder.andWhere('bookmark.listId IS NULL');
```

Comment:
Doesn't this battle the query bellow actually?

Reply:
Should be not. This one was wrapped with isPlus and the other one is the opposite 🙏

Reply:
Updated it to make the conditions simpler 🥳